engine/access/ingestion2/collection_syncer.go

```diff
@@ -405,17 +449,17 @@ func (s *CollectionSyncer) findLowestBlockHeightWithMissingCollections(
 }
 
 // OnCollectionDownloaded indexes and persists a downloaded collection.
-// This is a callback intended to be used with the requester engine.
-func (s *CollectionSyncer) OnCollectionDownloaded(_ flow.Identifier, entity flow.Entity) {
-	collection, ok := entity.(*flow.Collection)
-	if !ok {
-		s.logger.Error().Msgf("invalid entity type (%T)", entity)
-		return
-	}
-
-	err := indexer.IndexCollection(collection, s.collections, s.transactions, s.logger, s.collectionExecutedMetric)
+// This function is a callback intended to be used by the requester engine.
+//
+// Note: You might wonder what happens if the ingestion engine crashes,
+// but the requester engine continues to invoke this callback and writes
+// to a closed channel. In our current design, we intentionally ignore such cases.
+//
+// If any engine crashes, we treat it as a sign of possible state corruption,
+// and the node is expected to restart. Thus, this edge case is not handled explicitly.
+func (s *CollectionSyncer) OnCollectionDownloaded(id flow.Identifier, entity flow.Entity) {
+	err := s.pendingCollectionsHandler.Process(id, entity)
 	if err != nil {
-		s.logger.Error().Err(err).Msg("could not index collection after it has been downloaded")
-		return
+		s.logger.Warn().Err(err).Msg("failed to process pending collections")
```

Comment:
this is still using the log and continue pattern that we're trying to avoid. Typically, this is done within a `Process` method which returns an error. Since `OnCollectionDownloaded` is a `requester.HandleFunc` which does not return an error, we should just log a fatal here instead with a comment.
```suggestion
		// this is an unexpected error condition. The only expected error returned from Process
		// is for an unexpected type. since OnCollectionDownloaded is called from the requester
		// engine which is configured to only process collections, any error returned here indicates
		// a bug or state corruption.
		s.logger.Fatal().Err(err).Msg("failed to process pending collections")
		return
```